### PR TITLE
Generate the changelog at release time rather than updating in each PR #changelog

### PR DESCRIPTION
**Pull request by Adlai-Holler**
This reduces merge conflicts and standardizes changelog format.

If you _do_ change the changelog, now you get a warning unless you add \#changelog to the title.

To generate the changelog, just install https://github.com/github-changelog-generator/github-changelog-generator and run it in the project folder.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR modifies CI file</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/a33dfa7ee29e93f4b244f0e3eda1b519/raw/72cedc64bd3a2b678e9cca6040c88f2a5357e756/log.txt)

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR modifies CI file</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/25f2f0be2b032281070cd60e8cb6d139/raw/e2605184f64849627fb17a5a65f95d4ec62eb765/log.txt)

**Comment by Adlai-Holler** [comment]
CI is still broken @garrettmoon

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/f5960a7467335cbccbb656f5677cc586/raw/95e824443a0a2d139a4ee72feb47c1defd9196d6/log.txt)

**Comment by maicki** [comment]
@Adlai-Holler @nguyenhuy @appleguy @garrettmoon , I took this PR over and would like to move forward with it. I updated it so it's in a mergable state. I would appreciate your opinion and / or review of this PR.

Currently, most of the times something lands on master with a changelog change, all PR's that also include a changelog change have to resolve a merge conflict. Just resolving the changelog conflict will trigger a CI run with most of the time build approval involved. Generating the changelog on release time will remove this step from the PR process. Furthermore this standardize the changelog format.

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+### Preparation
+- Install [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator): `sudo gem install github_changelog_generator`
+- Generate a GitHub Personal Access Token to prevent running into public GitHub API rate limits: https://github.com/github-changelog-generator/github-changelog-generator#github-token
+
+### Process
+- Run `github_changelog_generator` in Texture project directory: `github_changelog_generator --token <generated personal token> TextureGroup/Texture`
+- Update `spec.version` within `Texture.podspec`.
+- Create a new PR with the updated `Texture.podspec` and the newly generated changelog, add `#changelog` to the PR message so the CI will not prevent merging it.
+- After merging in the PR. Create a tag at that commit and push to `TextureGroup/Texture`.
```
Comment (nguyenhuy) [inline RELEASE.md]:
```suggestion
- After merging in the PR, [create a new GitHub release](https://github.com/TextureGroup/Texture/releases/new).
- Publish the release to CocoaPod: `pod trunk push` (you'll need permissions for this).
- Announce the release in the slack channel.
```

**Pull request merged**